src/libelektra/kdb.c

```diff
@@ -677,6 +694,11 @@ static int elektraSetPrepare(Split *split, Key *parentKey, Key **errorKey)
 
 				if(p == 0)
 				{
+					if(hook)
+					{
+						ksRewind(split->keysets[i]);
```

Comment:
Can you document the behaviour, why execute the hook for every single mountpoint?

The global plugins rather should always have the whole configuration combined. If its impossible for this hook, it should be documented.